docs/reference/package-control/repository.md

```diff
@@ -0,0 +1,757 @@
+## `packages`
+
+The only required properties for each package are:
+
+- `details`
+- `releases`
+
+The `details` field is a GitHub, Bitbucket or GitLab repository URL (HTTPS).
+Properties of the package will be retrieved from that automatically,
+but can also be explicitly set here:
+
+- `name`<br>
+  For when the package name is different from the Git repository name.
+- `author` (either a string or an array of strings)<br>
+  For when it is different from the GitHub username owning the repository.
+- `description`<br>
+  For when it should be different from the GitHub repository description.
+- `readme`<br>
+  If different from each platform's default README URL.
+  This URL should be to the raw source of the file, not the rendered webpage.
+- `issues`<br>
+  When using a different bug tracker from each platform's default issues page.
+
+Additional optional URLs can be provided,
+which will be displayed on the Package Control website:
+
+- `homepage`<br>
+  For when the project has a website other than its repository.
+- `donate`<br>
+  For users to donate to the package maintainer.
+- `buy`<br>
+  For commercial packages.
+
+When [renaming a package][rename], existing installations can be "redirected" using this property:
+
+- `previous_names`
+
+To help users find a package, they can be tagged with [labels][]:
+
+- `labels`
+
+[rename]: renaming.html
+[labels]: submitting.html#what-labels-to-use
+
+
+### Quick examples
+
+This is all you need, with all the information pulled from the GitHub, Bitbucket or GitLab repository.
+
+```json
+{
+	"packages": [		
+		{
+			"details": "https://github.com/wbond/sublime_alignment",
+			"releases": [
+				{
+					"sublime_text": "*",
+					"tags": true
+				}
+			]
+		}
+	]
+}
+```
+
+This is more complete, and adds
+a specific name, labels and homepage and donation links.
+
+```json
+{
+	"packages": [
+		{
+			"name": "Solarized Color Scheme",
+			"details": "https://github.com/braver/Solarized",
+			"homepage": "https://ethanschoonover.com/solarized/",
+			"donate": "https://paypal.me/koenlageveen",
+			"labels": ["color scheme"],
+			"releases": [
+				{
+					"sublime_text": ">=3000",
+					"tags": true
+				}
+			]
+		}
+	]
+}
+```
+
+You can find tons of examples in the default repository
+at https://github.com/wbond/package_control_channel/tree/master/repository.
+
+
+### Package metadata without `details`
+
+Package metadata are resolved using `details`,
+where each field can be overridden by explicitly set values.
+If `details` is absent, each release must specify a `base` url.
+In this and other aspects,
+release based packages behave the same as tag based packages.
+
+```json
+{
+	"packages": [
+		{
+			"name": "Package With Asset",
+			"description": "A library with custom download assets",
+			"author": "Jon Doe",
+			"releases": [
+				{
+					"base": "https://github.com/SublimeText/PackageWithAsset",
+					"asset": "Package.With.Asset-${version}-st${st_build}-${platform}.sublime-package",
```

Comment:
"asset" is actually not supported by neither the new nor the old crawler.

Reply:
It is part of schema 4.0.0 and must be supported, however. Omitting it here is the wrong direction.

Reply:
There are a couple of features in here that are not supported or simply not allowed, on the public default channel. It makes sense the crawler for that channel doesn’t support the entire schema. I’ll add a note/caveat about the scope of this feature. Let me know if you spot any others.

Reply:
I disagree. The crawler has to fully support v4 schema to make use of all of its features as soon as we switch to the new site. That's the main point of all of this. Otherwise we can stop here.

Reply:
There is a good chance I’m misunderstanding things, so sorry for that. It’s my understanding that the crawler exclusively crawls the public default channel. We disallow certain constructions there, so the crawler doesn’t need to support them right? Or, hm, now that I think about it, probably some of the repositories included in the channel _can_ use those features… so even if those are edge cases or currently not in use, the crawler should support them… hmmmm!

Anyway, this is not a discussion about the schema or the design of the crawler. I don’t have opinions on either. The document should document the schema primarily of course. My intention was mostly to add something like this:
“If you’re looking to add a package to the default channel, you would not use assets.”.